storage/pebble/registers.go

```diff
@@ -79,17 +89,55 @@ func (s *Registers) Store(
 		return fmt.Errorf("failed to commit batch: %w", err)
 	}
 
+	// increment height if necessary
+	if height > s.latestHeight {
+		err = s.updateLatestHeight(height)
+		if err != nil {
+			return err
+		}
+	}
 	return nil
 }
 
-// TODO: Finish implementation after deciding prefixes and/or badger use
-
 // LatestHeight Gets the latest height of complete registers available
 func (s *Registers) LatestHeight() (uint64, error) {
-	panic("not implemented")
+	return s.heightLookup(latestHeightKey())
 }
 
 // FirstHeight first indexed height found in the store, typically root block for the spork
 func (s *Registers) FirstHeight() (uint64, error) {
-	panic("not implemented")
+	return s.heightLookup(firstHeightKey())
+}
+
+func (s *Registers) heightLookup(key []byte) (uint64, error) {
+	res, closer, err := s.db.Get(key)
+	defer closer.Close()
+	if err != nil {
+		return 0, err
+	}
+	return binary.BigEndian.Uint64(res), nil
+}
+
+func (s *Registers) indexHeightWithKey(key []byte, height uint64) error {
+	payload := make([]byte, 0)
+	encoded := binary.BigEndian.AppendUint64(payload, height)
+	return s.db.Set(key, encoded, nil)
+}
+
+func (s *Registers) updateFirstHeight(height uint64) error {
```

Comment:
this doesn't seem to be called anywhere. Should it be called in `Store` if `firstHeight` is unset?

Reply:
Also, I think this function should prevent overwriting the `firstHeight`. It could first read the db to see if a value exists and return an error if it does.